### PR TITLE
Implemented basic Cmake System

### PR DESCRIPTION
**Pull request by vpaladino778**
Moved .cpp into `sources/` folder and .h files into `include/` folder

Created basic CMakeLists.txt

Created build instructions in readme

Related issues: #3 #2

### CODE REVIEW

**Comment by vpaladino778** [comment]
This build fails with 

```Bash
In file included from /mnt/c/Users/Vincent/Desktop/Repositories/colline/code/src/Agent.cpp:6:0: 
/mnt/c/Users/Vincent/Desktop/Repositories/colline/code/include/stdafx.h:15:10: fatal error: afx.h:
No such file or directory
 #include <afx.h> 
```

According to [this](https://stackoverflow.com/questions/16227840/where-can-i-get-afx-h-files) AFX is the "Application Framework Extensions" which is the original version of Microsoft Foundation Classes.

**Comment by s-martin** [comment]
Yeah, the project relies heavily on MFC (also for GUI portions), which should be removed to provide platform independence, see also #7 and #8

**Comment by vpaladino778** [comment]
> Yeah, the project relies heavily on MFC (also for GUI portions), which should be removed to provide platform independence, see also #7 and #8

Is there anything i should change on this pull request before it can be merged in? I figure swapping out MFC should probably be another pull request

**Comment by s-martin** [comment]
The dependency on MFC would be needed in the CMake file for MSVC builds, but the question is, if this makes much sense, if MFC is removed later anyway.

And in the current state the project probably won't build under Linux/macOS anyway, because it makes heavy usage of MFC-only (i.e. MSVC only) data types.

**Comment by s-martin** [comment]
However, I'm not the founder of this project, so I won't make any decisions. Just providing my thoughts.

File: code/README.md [hunk 1]
```diff
@@ -32,6 +32,19 @@ StdAfx.h, StdAfx.cpp
 Resource.h
     This is the standard header file, which defines new resource IDs.
     Microsoft Visual C++ reads and updates this file.
+/////////////////////////////////////////////////////////////////////////////
+
```
Comment (s-martin) [inline code/README.md:36]: To be decided, if this readme file should be used. This is the standard readme Visual Studio creates for a new project, which provides not very much information.

From my point of view the current edits should be added to Readme.md in the root directory, so it's shown on the project page. This file here in the /code directory could be removed.

**Comment by vpaladino778** [comment]
@s-martin I'll merge this in for now, is it possible to include MFC for now and we can work on swapping it out for something more modern at a later time?

**Pull request merged**